### PR TITLE
Alu builtins

### PR DESCRIPTION
**Pull request by realqhc**
The builtins have been implemented, and it would be great if someone could explain a bit about what error messages are required for some of the builtins in xcvalu extension.

### CODE REVIEW

File: llvm/test/CodeGen/RISCV/corev/alu.ll [hunk 1]
```diff
@@ -496,7 +485,14 @@ declare i32 @llvm.riscv.cv.alu.addn(i32, i32, i32)
 define i32 @test.cv.alu.addn.case.a(i32 %a, i32 %b) {
 ; CHECK-LABEL: test.cv.alu.addn.case.a:
 ; CHECK:       # %bb.0:
-; CHECK-NEXT:    cv.addn a0, a0, a1, 15
+; CHECK-NEXT:    addi sp, sp, -16
```
Comment (ChunyuLiao) [inline llvm/test/CodeGen/RISCV/corev/alu.ll]: This shouldn't be right here, the generated assembly should still look the same as before.

**Comment by realqhc** [comment]
@jeremybennett @PaoloS02 can we get this commit merged soon?

File: clang/include/clang/Basic/BuiltinsRISCVCOREV.def [hunk 2]
```diff
@@ -177,6 +177,48 @@ TARGET_BUILTIN(bitmanip_cnt, "UZiUZi", "nc", "xcvbitmanip")
 TARGET_BUILTIN(bitmanip_ror, "UZiUZiUZi", "nc", "xcvbitmanip")
 TARGET_BUILTIN(bitmanip_bitrev, "UZiUZiIUcIUc", "nc", "xcvbitmanip")
 
+TARGET_BUILTIN(alu_slet,
+"ZiZiZi", "nc", "xcvalu")
```
Comment (PaoloS02) [inline clang/include/clang/Basic/BuiltinsRISCVCOREV.def]: Any reason for splitting the declarations in two lines like this?

File: llvm/lib/Target/RISCV/RISCVInstrInfoCOREV.td [hunk 3]
```diff
@@ -882,7 +882,6 @@ let Predicates = [HasExtXcvalu] in {
   def : Pat<(srl (add (sub GPR:$rd, GPR:$rs1), (roundBit GPR:$rs2)), GPR:$rs2),
             (CV_SUBURNR GPR:$rd, GPR:$rs1, GPR:$rs2)>;
 
-  def : PatCoreVAluGpr<"abs", "ABS">;
   def : PatCoreVAluGprGpr<"slet", "SLET">;
   def : PatCoreVAluGprGpr<"sletu", "SLETU">;
```
Comment (PaoloS02) [inline llvm/lib/Target/RISCV/RISCVInstrInfoCOREV.td]: Will need to change SLET to SLE and SLETU to SLEU when rebasing to solve the merge conflict

**Comment by PaoloS02** [comment]
I'm good to merge this as soon as we clarify the indentation matter and the conflicts are solved. One conflict is caused only by the addition of the alu and mac builtins at the "same time". They just need to be both kept. For the second conflict I left a comment.

**Pull request merged**